### PR TITLE
Feature/16 cdk create api gateway and cognito

### PR DESCRIPTION
**Pull request by taniguchiiqqq**
## Overviews of implementation
add the following resources.
- Cognito
  - UserPool
  - UserClient
  - IdentityProvider(Google)
- API Gateway
  - RestApi
  - Authenticator
 
## Review points
- where we should have the congnitoDomainPrefix value. In ParameterStore or props on the source code.
- this code works for deploy.

Now, I set up the API Gateway with only basic options. so we should change it on another issue.

### CODE REVIEW

File: cdk/test/nishikiBackend-stack.test.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import * as cdk from "aws-cdk-lib";
+import { Template } from "aws-cdk-lib/assertions";
+import { NishiliBackendStack } from "../lib/nishikiBackend-stack";
+
+describe("NishiliBackendStack", () => {
+  test("sysnthesizes correctly", () => {
```
Comment (hitohata) [inline cdk/test/nishikiBackend-stack.test.ts:6]: this name may be typo

**Review by hitohata — APPROVED** [review]
> where we should have the congnitoDomainPrefix value. In ParameterStore or props on the source code.

The test failed. Do you mean, I need to set the above value when running the test or do you have any idea how to pass the test?

But at this time, it's not important to pass a test. So, I approve this PR.

P.S.
I think the `cdk.test.ts` file is no longer needed.

File: cdk/lib/nishikiBackend-stack.ts [hunk 2]
```diff
@@ -1,10 +1,169 @@
...
+        minLength: 8,
+        requireLowercase: true,
+        requireUppercase: true,
+        requireDigits: true,
+      },
+      accountRecovery: AccountRecovery.EMAIL_ONLY,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+    });
+
+    const congnitoDomainPrefix = ssm.StringParameter.valueForStringParameter(
```
Comment (hitohata) [inline cdk/lib/nishikiBackend-stack.ts:41]: typo Cognito

**Comment by taniguchiiqqq** [comment]
@hitohata 
> The test failed. Do you mean, I need to set the above value when running the test or do you have any idea how to pass the test?

I do appreciate your comment. That's because I commented in some code about RestApi and the Authorizer must be attached to one RestApi. You don't need some specific operation to pass the test now.

But if you want to deploy it to your AWS account, you have to follow this steps.
1. create the following parameter in AWS paramater store. "/nishiki/prod/cognito-domain-prefix"= "any string"
2. create the Google OAuth client id and secret. ( [document](https://developers.google.com/identity/sign-in/web/sign-in) )   
  2-1.  Authorized domain = amazoncognito.com
  2-2. Approved JavaScript origins = https://${cognitoDomainPrefix}.auth.${region}.amazoncognito.com
  2-3 Approved redirect URI = http://localhost
3. create the following parameters in AWS paramater store.
  3-1. "/nishiki/prod/google-client-id" = client id issued by google
  3-2. "/nishiki/prod/google-client-secret" = client secret issued by google
4. run `cdk deploy`

**Pull request merged**